### PR TITLE
Middleware support for asynchronous requests.

### PR DESCRIPTION
**Pull request by jaydensmith**
The middleware was causing some issues when using async views due to not supporting async requests.

### CODE REVIEW

File: django_pgschemas/middleware.py [hunk 1]
```diff
@@ -89,6 +94,16 @@ def middleware(request):
...
+
+    if iscoroutinefunction(get_response):
+        async_logic = sync_to_async(logic)
+
+        async def middleware(request):
+            await async_logic(request)
+            return await get_response(request)
+    else:
+        def middleware(request):  # type: ignore
+            logic(request)
```
Comment (lorinkoz) [inline django_pgschemas/middleware.py:106]: Please double check the logic here. `logic` can also return a redirection (in the middle of the function body) and therefore the function is not extractable as you have it.

Comment (lorinkoz) [reply django_pgschemas/middleware.py:106]: The failing tests are for this reason.

**Review by lorinkoz — CHANGES_REQUESTED** [review]
Thank you for the fix!

File: django_pgschemas/middleware.py (hunk 1 above)
Comment (jaydensmith) [reply django_pgschemas/middleware.py]: @lorinkoz thanks for reviewing, I'm unable to run all the tests unfortunately, I get an `AppRegistryNotReady`. I've just pushed a change that should handle the redirect, however.

**Review by lorinkoz — APPROVED** [review]
Thanks! 🚀

**Comment by lorinkoz** [comment]
@jaydensmith v0.15.1 is out, thanks again!

**Comment by lorinkoz** [comment]
If you wanna take a look, this is the port of your code to `master` with some tests: https://github.com/lorinkoz/django-pgschemas/pull/183

**Pull request merged**